### PR TITLE
EXPERIMENTAL_TABLEV2 Line Actions

### PR DESCRIPTION
**Pull request by fanny**
#### What is the purpose of this pull request?

as the title says

#### What problem is this solving?
This feature creates a last extra column with an ActionMenu component per line.

#### How should this be manually tested?
Clone the repo and `yarn install`, `yarn start`

#### Screenshots or example usage
image

#### Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [X] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/EXPERIMENTAL_Table/LineActions/index.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React, { FC } from 'react'
+
+import ActionMenu from '../../ActionMenu'
+import OptionsDots from '../../icon/OptionsDots'
+
+const LineAction: FC<LineActionProps> = ({ lineActions, rowData }) => (
+  <>
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Table/LineActions/index.tsx]: Is that fragment really necessary?

File: react/components/EXPERIMENTAL_Table/LineActions/index.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        ...action,
+        label: action.label({ rowData }),
+        onClick: () => action.onClick({ rowData }),
+      }))}
+    />
+  </>
+)
+
+export type LineActionObject = {
+  label: Function,
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Table/LineActions/index.tsx]: Is that semantic correct? 🤔In my mind, a label should be a string.

File: react/components/EXPERIMENTAL_Table/hooks/useTableLineActions.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+const NO_TITLE_COLUMN = ' '
+const LINE_ACTION_ID = 'lineAction'
+
+const useTableLineActions = ({
+  items,
+  columns,
+  lineActions
+}: hookInput): hookReturn => {
+  
+  const withLineActionItems = useMemo<Array<Object>>(() => {
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Table/hooks/useTableLineActions.tsx]:
```suggestion
  const itemsWithLineActions = useMemo<Array<Object>>(() => {
```
I think that is a better naming 🤔

File: react/components/EXPERIMENTAL_Table/hooks/useTableLineActions.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    return lineActions ? items.map((item) => ({ lineAction: true, ...item })): items
+  }, [items])
+
+  const withLineActionColumns = useMemo<Array<Column>>(() => {
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Table/hooks/useTableLineActions.tsx]:
```suggestion
  const columnsWithLineActions = useMemo<Array<Column>>(() => {
```

File: react/components/EXPERIMENTAL_Table/LineActions/index.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React, { FC } from 'react'
+
+import ActionMenu from '../../ActionMenu'
+import OptionsDots from '../../icon/OptionsDots'
+import { render } from 'react-dom';
```
Comment (vitoriaHeliane) [inline react/components/EXPERIMENTAL_Table/LineActions/index.tsx]: Remove the `;`

File: react/components/EXPERIMENTAL_Table/hooks/useTableLineActions.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      ...columns,
+      {
+        id: LINE_ACTION_ID,
+        title: NO_TITLE_COLUMN,
+        cellRender,
+      },
+    ]: columns
+  }, [columns])
+
+  console.log(columnsWithLineActions);
```
Comment (vitoriaHeliane) [inline react/components/EXPERIMENTAL_Table/hooks/useTableLineActions.tsx]: Is it necessary?

File: react/components/EXPERIMENTAL_Table/LineActions/index.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+      label: action.renderLabel? action.renderLabel({ rowData }): action.label,
+      onClick: () => action.onClick({ rowData }),
+    }))}
+  />
+)
+
+export type LineActionObject = {
+  label: String,
+  renderLabel?: ({ rowData: unknow}) => React.ReactNode,
+  isDangerous: Boolean,
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Table/LineActions/index.tsx]:
```suggestion
  isDangerous?: boolean,
```

File: react/components/EXPERIMENTAL_Table/LineActions/index.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    options={lineActions.map(action => ({
+      ...action,
+      label: action.renderLabel? action.renderLabel({ rowData }): action.label,
+      onClick: () => action.onClick({ rowData }),
+    }))}
+  />
+)
+
+export type LineActionObject = {
+  label: String,
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Table/LineActions/index.tsx]:
```suggestion
  label: string,
```

**Pull request merged**